onnxruntime/hosting/http/http_server.h

```diff
@@ -34,14 +40,14 @@ class App {
 
   App& Bind(net::ip::address address, unsigned short port);
   App& NumThreads(int threads);
+  App& OnStart(const start_fn& fn);
   App& Post(const std::string& route, const handler_fn& fn);
```

Comment:
NIT: consider change it to `RegisterPost` to make it clear?

Reply:
Yeah I'm not happy about the naming. Maybe `OnPost`?

Reply:
Going with `OnStartup` and `OnPost`

Reply:
Usually `OnFoo` means an action to do the Foo action. But here we literally only register the action, not take the action.

Reply:
To me `OnFoo` means execute this action when `Foo` occurs. In this case, register the POST method and execute when a POST request occurs.

Reply:
Usually one function only do one thing and seems different people having different understanding of `OnFoo`. What if we make the name clear that this is only a registration but not execution?

Reply:
Fine with me. Done!